src/pdf/ShoppingList/dataPreparation.ts

```diff
@@ -0,0 +1,247 @@
+import { Person } from "@/components/Form/formFunctions";
+
+export interface BlockProps {
+    [key: string]: string;
+}
+
+interface ParcelInfoAndClientID {
+    parcelInfo: ParcelInfo;
+    clientID: string;
+}
+
+interface ClientDataAndFamilyData {
+    clientData: Schema["clients"];
+    familyData: Schema["families"][];
+}
+
+interface NappySizeAndExtraInformation {
+    nappySize: string;
+    extraInformation: string;
+}
+
+export interface ShoppingListPDFProps {
+    postcode: string;
+    parcelInfo: ParcelInfo;
+    clientSummary: ClientSummary;
+    householdSummary: HouseholdSummary;
+    requirementSummary: RequirementSummary;
+    itemsList: Item[];
+    endNotes: string;
+}
+
+export interface ParcelInfo extends BlockProps {
+    voucherNumber: string;
+    packingDate: string;
+    collectionDate: string;
+    collectionSite: string;
+}
+
+export interface ClientSummary {
+    name: string;
+    contact: string;
+    address: string;
+    extraInformation: string;
+}
+
+export interface HouseholdSummary extends BlockProps {
+    householdSize: string;
+    genderBreakdown: string;
+    numberOfBabies: string;
+    ageAndGenderOfChildren: string;
+}
+
+export interface RequirementSummary extends BlockProps {
+    feminineProductsRequired: string;
+    babyProductsRequired: string;
+    petFoodRequired: string;
+    dietaryRequirements: string;
+    otherItems: string;
+}
+
+export interface Item {
+    description: string;
+    quantity: string;
+    notes: string;
+}
+
+const formatDate = (dateString: string | null): string => {
+    if (!dateString) {
+        return "";
+    }
+    return new Date(dateString).toLocaleString("en-GB", {
+        year: "numeric",
+        month: "numeric",
+        day: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+    });
+};
+
+const prepareParcelInfo = async (parcelID: string): Promise<ParcelInfoAndClientID> => {
+    const fetchedData = await fetchParcels(parcelID);
+    const parcelInfo: ParcelInfo = {
+        voucherNumber: fetchedData.voucher_number ?? "",
+        packingDate: formatDate(fetchedData.packing_datetime),
+        collectionDate: formatDate(fetchedData.collection_datetime),
+        collectionSite: fetchedData.collection_centre ?? "",
+    };
+
+    if (parcelInfo.collectionSite === "Delivery") {
+        parcelInfo.collectionSite = "N/A - Delivery";
+    }
+
+    return { parcelInfo: parcelInfo, clientID: fetchedData.client_id };
+};
+
+const getClientAndFamilyData = async (clientID: string): Promise<ClientDataAndFamilyData> => {
+    const clientData = await fetchClients(clientID);
+    const familyData = await fetchFamilies(clientData.family_id);
+
+    return { clientData: clientData, familyData: familyData };
+};
+
+const processExtraInformation = (original: string): NappySizeAndExtraInformation => {
+    if (original.startsWith("Nappy Size: ")) {
+        const [nappySize, extraInformation] = original.split(", Extra Information: ");
```

Comment:
`.split("...", 1)` maybe is better because then if there are multiple of these strings only the first is split out